MapboxNavigation/NavigationMapView.swift

```diff
@@ -671,6 +676,188 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
     }
 
     // MARK: - Vanishing route line methods
+
+    private var primaryRoutePoints: RoutePoints?
+    private var primaryRouteLineGranularDistances: RouteLineGranularDistances?
+    private var primaryRouteRemainingDistancesIndex: Int?
+    public var newFractionTraveled: Double = 0.0
+    
+    private struct RoutePoints {
```

Comment:
Is it necessary to store both `nestedList` and `flatList`? Would it be possible to store only list of coordinates?

Reply:
I think storing the `nestedList` would be used in the  `func updateUpcomingRoutePointIndex(routeProgress:)` , because we'll need the `nestedList[routeProgress.legIndex]` to find out the remaining coordinates number based on the route progress updates. Otherwise, the route, route leg, route step, if we flatten them all, it's hard to track them when there's any route progress update with the index.